test/check-application

```diff
@@ -1549,10 +1549,8 @@ class TestFiles(testlib.MachineCase):
         self.enter_files()
 
         # Copy/paste file
-        m.execute("""
-            runuser -u admin mkdir /home/admin/newdir
-            runuser -u admin echo "test_text" > /home/admin/newfile
-        """)
+        m.execute("runuser -u admin mkdir /home/admin/newdir")
```

Comment:
Ah -- `m.write()` actually does `mkdir` by itself, but it doesn't apply the `owner` argument to it. That's something to fix.. (But nevermind, this is good enough!)

Reply:
This is actually a difficult nut to crack: we currently use a simple unconditional `mkdir -p`.  So: if the directory doesn't already exist we want to create it with the correct permissions.  But if it does already exist, we don't want to change its permissions, probably.  We'd probably have to up our game here and figure out which components got created and chown those.  Note also that `runuser mkdir` is probably not what we want either: it could well be the case that the parent directories aren't writable by that user.

And what of the permissions kwarg?  Probably we don't want to apply those to the directory....